snuba_sdk/mql/mql.py

```diff
@@ -168,21 +176,40 @@ def visit_term(
         Checks if the current node contains two term children, if so
         then merge them into a single Formula with the operator.
         """
-        coefficient_left, zero_or_more_others = children
-        assert isinstance(coefficient_left, (Formula, Timeseries, float, int, str))
+        unary_left, zero_or_more_others = children
+        assert isinstance(unary_left, (Formula, Timeseries, float, int, str))
 
         if zero_or_more_others:
             for zero_or_more in zero_or_more_others:
-                _, term_operator, _, coefficient_right, *_ = zero_or_more
-                coefficient_left = Formula(
-                    term_operator, [coefficient_left, coefficient_right]
-                )
+                _, term_operator, _, unary_right, *_ = zero_or_more
+                unary_left = Formula(term_operator, [unary_left, unary_right])
 
-        return cast(Union[Formula, Timeseries, float, int, str], coefficient_left)
+        return cast(Union[Formula, Timeseries, float, int, str], unary_left)
 
     def visit_term_op(self, node: Node, children: Sequence[Any]) -> Any:
         return TERM_OPERATORS[node.text]
 
+    def visit_unary(
+        self, node: Node, children: Sequence[Any]
+    ) -> Union[Formula, Timeseries, float, int, str]:
+        unary_op, coefficient = children
+        if unary_op:
+            if isinstance(coefficient, float) or isinstance(coefficient, int):
+                return -coefficient
+            elif isinstance(coefficient, Formula) or isinstance(
+                coefficient, Timeseries
+            ):
+                return Formula(function_name=unary_op[0], parameters=[0, coefficient])
```

Comment:
Why does this have two parameters? Is that to support `minus()`? If so, we should use `negate()`.

Reply:
Oh I didn't know there was that function, will use that!